lib/compress/zstd_fast.c

```diff
@@ -59,9 +60,41 @@ size_t ZSTD_compressBlock_fast_generic(
     U32 offset_1=rep[0], offset_2=rep[1];
     U32 offsetSaved = 0;
 
+    /* This is all complicated by the fact that we need to handle positions
+     * specified in 3 different ways: by direct pointers, by indices relative
```

Comment:
just for my memory : in which case are positions tracked by direct pointers only ?
I thought everything was using indices, although relative to up to 2 different bases.

Reply:
Right, no data structure tracks things by pointers. But pointers are used in this function for more than just accesses, they're compared, etc. That's all I'm saying. I guess this is really a to-do directed at me to see if I can clean all this up some.